candi/bashible/common-steps/node-group/064_configure_kubelet.sh.tpl

```diff
@@ -235,4 +245,6 @@ containerLogMaxSize: {{ .nodeGroup.kubelet.containerLogMaxSize | default "50Mi"
 containerLogMaxFiles: {{ .nodeGroup.kubelet.containerLogMaxFiles | default 4 }}
 {{- end }}
 allowedUnsafeSysctls:  ["net.*"]
+shutdownGracePeriod: ${shutdownGracePeriod}
+shutdownGracePeriodCriticalPods: ${shutdownGracePeriodCriticalPods}
```

Comment:
Shouldn't we check that Kubernetes version is greater or equal to 1.21?

Reply:
And also what about AWS node-termintation handler? Do we still need it if we use kubelet?

Reply:
It'll get ignored when deserializing to Go struct.

Yes, the shutdown timeout is unspecified in AWS. 4 minutes is a best effort guesstimate. With node-termination-handler we get additional guaranteed 2 minutes for Spot VMs. Not bad at all.